ava-labs/core-extension
# refactor: perform eth_sendTransaction via EVM module

* [CP-9159](https://ava-labs.atlassian.net/browse/CP-9159)
* _+ migrate `eth_sendTransaction` to be performed via EVM Module_

---

* Related PR: https://github.com/ava-labs/vm-modules/pull/160

## Changes
* Starts using the `updateTx` callback provided by VM Modules for updating transaction fee configs / spend limits (see related PR above)
* Removes `eth_sendTransaction` handler and some of the related files so that those requests are performed using the `EvmModule`
    * Also removes the `SignTransaction.tsx` file / route & related files
* Updates `useFeeCustomizer` hook to also handle EVM fees
* Updates approval screen components to the new `displayData` schema so they can be used on `GenericApprovalScreen`

## Testing
Basically a full regression on EVM features is needed:
* Send
* Bridge
* Swap
    * _all of the above features but initiated by Core Web_
* Send NFT
* dApp integrations (i.e. liquidity pool on TraderJoe.xyz)
    * make sure token spend approvals work as they used to
    * make sure LBT (Liquidity Book Tokens) collapse in an accordion
* Verify that the `chainId` parameter in `eth_sendTransaction` payload works as it used to
    * no cross-environment transaction execution allowed (no mainnet execution while UI is on testnet or vice-versa)
    * transactions on custom networks are only allowed if the custom network is also active in the UI


## Checklist for the author
